apps/src/code-studio/levels/dialogHelper.js

```diff
@@ -11,28 +11,29 @@ import Sounds from '../../Sounds';
  * This file contains general logic for displaying modal dialogs
  */
 
-var dialogType = null;
 var adjustedScroll = false;
 
-function dialogHidden() {
-  var lastServerResponse = window.dashboard.reporting.getLastServerResponse();
-  if (dialogType === "success" && lastServerResponse.nextRedirect) {
-    window.location.href = lastServerResponse.nextRedirect;
-  }
-
-  if (dialogType === "error") {
-    adjustScroll();
+/**
+ * @param {string|Component} typeOrComponent - Either a string identifying the DOM
+ *   to use in this dialog, or a ReactComponent representing the contents. Of the
+ *   two, the latter is preferable.
+ * @param {function} callback - Method to call when OK is clicked
+ * @param {function} onHidden - Method called when dialog is hidden/closed
+ */
+export function showDialog(typeOrComponent, callback, onHidden) {
+  let content;
+  if (typeof(typeOrComponent) === 'string') {
+    // Use our prefabricated dialog content.
+    content = document.querySelector("#" + typeOrComponent + "-dialogcontent").cloneNode(true);
+  } else {
+    const div = document.createElement('div');
+    ReactDOM.render(typeOrComponent, div);
+    content = div.childNodes[0];
   }
-}
-
-export function showDialog(type, callback, onHidden) {
-  dialogType = type;
-
-  // Use our prefabricated dialog content.
-  var content = document.querySelector("#" + type + "-dialogcontent").cloneNode(true);
-  var dialog = new LegacyDialog({
+  const dialog = new LegacyDialog({
+    // Content is a div with a specific expected structure. See LegacyDialog.
     body: content,
-    onHidden: onHidden || dialogHidden,
+    onHidden,
```

Comment:
See https://github.com/code-dot-org/code-dot-org/pull/19556/commits/7ce143d46bfa26df8784664129b0774dfeb7719e for this particular change.